internal/lock.go

```diff
@@ -149,11 +149,21 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	status_line := Status_Line{filteredResources, make(chan int)}
+
+	if bar {
+		status_line.run()
+	}
+
+	for i, r := range filteredResources {
 		resource := r
 		go func() {
+
 			err := resource.Download(dir, mode, ctx)
 			errorCh <- err
+
+			status_line.indexCh <- i
```

Comment:
Please encapsulate this into a method (`statusLine.increment()`). This way the call sites don't have to manipulate internal fields.